applications/GeoMechanicsApplication/python_scripts/geomechanics_solver.py

```diff
@@ -190,8 +187,7 @@ def GetMinimumBufferSize(self):
     def ImportModelPart(self):
         """This function imports the ModelPart
         """
-        if self.solver_imports_model_part:
-            self._ImportModelPart(self.main_model_part, self.settings["model_import_settings"])
+        self._ImportModelPart(self.main_model_part, self.settings["model_import_settings"])
```

Comment:
the idea is to use the "standard" Kratos importing mechanism

Reply:
actually i think that this would correspond the mechanism used in the base_analysis_stage, so prabably we could directly remove the reimplementation of this function

Reply:
> actually i think that this would correspond the mechanism used in the base_analysis_stage, so prabably we could directly remove the reimplementation of this function

We can't since the base Python solver throws an error in the  `ImportModelPart`.

What we can eventually consider in here is to substitute it by `pass`. However, this will enforce to always import the mesh at the `AnalysisStage` level, breaking retrocompatibility of already existent (not necessarily multistage) cases. Having said this, I'd keep it as it is right now, at least for a while. In the future we can throw a deprecation warning in the `_ImportModelPart` so we deprecate the mechanism at once for the whole Kratos.

Reply:
Please note that the suggested changes break the way we currently carry out staged analysis. As you may have noticed, this branch has two test failures. We feel that the more gentle approach may be to duplicate one of our multi-stage tests and change it such that is uses your multi-stage approach. In this way, we don't break our production code, but we do create an opportunity to get some hands-on experience with your multi-stage approach. What do you think?

I will try to create a simple test by myself (based on an existing multi-stage test), but I may or may not need a bit of help to get it working. When I do, I will contact you by email. Thanks.

Reply:
Sure @avdg81 (this is why we make it draft). We created the PR to have something to discuss on top of. I think that the best is to arrange a short meeting to discuss precisely about this, how to do the changes without breaking your current stuff.

Reply:
Sure :+1:

Reply:
@rubenzorrilla As far as I understand, we would need to migrate our multi-stage analysis tests to use the new workflow. This is something we could get started with (looking at @mcgicjn2 for a possible timeline there). We would need some time to communicate the change to users of the geomechanics application and deprecate the old mechanism.

Another question from our side is if the changes in kratos related to the orchestrator could have an influence on a C++-only workflow we have, which uses a similar multi-stage approach as these failing python tests? For this specific PR, there are no risks since it's python-only, but are there any other changes planned in the C++ core?

Possibilities for having (part of) this powerful concept of orchestrator-based analysis available from C++ would be a separate discussion for later, I can imagine.

Reply:
> @rubenzorrilla As far as I understand, we would need to migrate our multi-stage analysis tests to use the new workflow. This is something we could get started with (looking at @mcgicjn2 for a possible timeline there). We would need some time to communicate the change to users of the geomechanics application and deprecate the old mechanism.
> 

Very good. I'll try to recover the geomechanics use case I used during the development just in case it is of help for this. Besides, I would also consider @avdg81 suggestion of temporarily duplicating the tests (keep current ones and add new ones with the standard mechanism).

> Another question from our side is if the changes in kratos related to the orchestrator could have an influence on a C++-only workflow we have, which uses a similar multi-stage approach as these failing python tests? For this specific PR, there are no risks since it's python-only, but are there any other changes planned in the C++ core?
> 

No. Multistage is conceived to occur at Python level only. Having said this, I think that your current C++ workflow shouldn't be affected by the new one.  

> Possibilities for having (part of) this powerful concept of orchestrator-based analysis available from C++ would be a separate discussion for later, I can imagine.

Yes. At some point we spoke with @mcgicjn2 about this. The idea is that the official multistage must remain at the Python level. Nevertheless, and considering your current needs, we can give you support on creating your custom C++ implementation emulating the Python official one.

Reply:
Clear, thank you! Indeed it's probably best to temporarily have the two mechanism both working correctly in this transition period (if this is possible). We'll add some issues to our board, which we can pick up after prioritization by @mcgicjn2!